glue/utils/matplotlib.py

```diff
@@ -331,6 +333,41 @@ def cache_axes(axes, toolbar):
     return cache
 
 
+class ColormapPatchHandler(HandlerBase):
+    def __init__(self, cmap, nb_subpatch=10, xpad=0.0, ypad=0.0):
+        """
+        A custom legend handler to represent 2D datased coded in colormaps
+
+        Parameters
+        ----------
+        cmap : `~matplotlib.colors.colormap`
+            The matplotlib colormap to use
+        nb_subpatch : int, optional
+            The number of stripes to use to represent the colormap.
+            The default is 10.
+        xpad : float, optional
+            Padding in the x direction. The default is 0.0.
+        ypad : float, optional
+            Padding in the y direction. The default is 0.0.
+
+        """
+        super().__init__(xpad, ypad)
+        self.nb_subpatch = nb_subpatch
+        self.cmap = cmap
+
+    def create_artists(self, legend, orig_handle,
+                       xdescent, ydescent, width, height, fontsize,
+                       trans):
+        collection = []
+        for i in range(self.nb_subpatch):
+            width_sub = width / self.nb_subpatch
+            x = xdescent + i * width_sub
+            collection.append(
+                Rectangle((x, ydescent), width_sub, height, transform=trans,
+                          facecolor=self.cmap(i / (self.nb_subpatch - 1)),
+                          edgecolor="none"))
```

Comment:
I've looked into it, and for me the main problem is that a colorbar needs an Axes to be drawn into it.
But the legend (if I understand correctly) is not a Axes, but an Artist added on top of an Axes.

So it will make the positioning a bit tricky, especially since matplotlib is in charge of positioning the legend.